packages/assertion-monitor/index.ts

```diff
@@ -0,0 +1,200 @@
+
+    const result = await processChunk(currentFromBlock, currentToBlock, client)
+    results.push(result)
+
+    if (currentToBlock === toBlock) break
+
+    currentFromBlock = currentToBlock + 1n
+  }
+
+  return results
+}
+const getBlockRange = async (
+  client: PublicClient,
+  childChainInfo: ChainInfo
+) => {
+  const latestBlockNumber = await client.getBlockNumber()
+  const toBlock = BigInt(options.toBlock) || latestBlockNumber
+
+  if (options.fromBlock) {
+    return { fromBlock: BigInt(options.fromBlock), toBlock }
+  }
+
+  const blockTime = getDefaultBlockRange(
+    getChainFromId(childChainInfo.parentChainId)
+  )
+
+  const fromBlock = await client.getBlock({
+    blockNumber: latestBlockNumber - BigInt(blockTime),
+  })
+
+  return { fromBlock: fromBlock.number, toBlock }
+}
+
+const monitorNodeCreatedEvents = async (childChainInfo: ChainInfo) => {
+  const parentChain = getChainFromId(childChainInfo.parentChainId)
+  const client = createPublicClient({
+    chain: parentChain,
+    transport: http(childChainInfo.parentRpcUrl),
+  })
+
+  const { fromBlock, toBlock } = await getBlockRange(client, childChainInfo)
+
+  const getLogsForChunk = async (
+    chunkFromBlock: bigint,
+    chunkToBlock: bigint,
+    client: PublicClient
+  ) => {
+    return client
+      .getLogs({
+        address: childChainInfo.ethBridge.rollup as `0x${string}`,
+        event: nodeCreatedEventAbi,
+        fromBlock: chunkFromBlock,
+        toBlock: chunkToBlock,
+      })
+      .catch(console.error)
+  }
+
+  const logsArray = await processChunkedRange(
+    fromBlock,
+    toBlock,
+    CHUNK_SIZE,
+    client,
+    getLogsForChunk
+  )
+
+  const logs = logsArray.flat()
+
+  const isDefaultSettings = !options.fromBlock && !options.toBlock
+  const durationString = isDefaultSettings
+    ? 'in last 7 days'
+    : `from block ${fromBlock} to block ${toBlock}`
+
+  const latestSafeBlock = await client.getBlock({ blockTag: 'safe' })
+
+  const latestSafeBlockTimestamp =
+    new Date(Number(latestSafeBlock.timestamp) * 1000).toLocaleString() + ' UTC'
+
+  const isLatestSafeBlockWithinRange =
+    latestSafeBlock.number < toBlock && latestSafeBlock.number > fromBlock
+
+  if (!logs || logs.length === 0) {
+    return {
+      chainName: childChainInfo.name,
+      alertMessage: `No assertion events found on ${
+        childChainInfo.name
+      } ${durationString}. Latest batch ${
+        isLatestSafeBlockWithinRange ? 'was' : 'was not'
+      } posted within this duration, at ${latestSafeBlockTimestamp} (block ${
+        latestSafeBlock.number
+      })`,
+    }
+  } else {
+    console.log(
+      `Found ${logs.length} assertion events on ${childChainInfo.name} ${durationString}.`
+    )
+    return null
+  }
+}
+
+const main = async () => {
+  try {
+    const alerts: { chainName: string; alertMessage: string }[] = []
+
+    for (const chainInfo of config.childChains) {
+      const result = await monitorNodeCreatedEvents(chainInfo)
```

Comment:
Let's also log which chain is being processed in the loop. Currently while running the monitor - I have to wait a minute or two of no activity to get to know the script is running. Being more explicit about the process in logs is a good dev exp.

Reply:
+1 to this